hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -1862,6 +1863,53 @@ private class Emit(
         }
         val moveArgs = refs.map( ref =>  Code(ref.m := ref.tempM, ref.v.storeAny(ref.tempV)) )
         EmitTriplet(Code(Code(storeTempArgs ++ moveArgs: _*), coerce[Unit](jump)), const(false), Code._empty)
+      case x@ReadValue(path, spec, requestedType) =>
+        val p = emit(path, env)
+        val pathString = path.pType2.asInstanceOf[PString].loadString(p.value[Long])
+        val rowBuf = spec.buildCodeInputBuffer(mb.fb.getUnsafeReader(pathString, true))
+
+        val (_, dec) = typeToTypeInfo(x.pType2) match {
+          case BooleanInfo => spec.buildEmitDecoderF[Boolean](requestedType, mb.fb)
+          case IntInfo => spec.buildEmitDecoderF[Int](requestedType, mb.fb)
+          case LongInfo => spec.buildEmitDecoderF[Long](requestedType, mb.fb)
+          case FloatInfo => spec.buildEmitDecoderF[Float](requestedType, mb.fb)
+          case DoubleInfo => spec.buildEmitDecoderF[Double](requestedType, mb.fb)
+        }
+        EmitTriplet(p.setup, p.m, dec(er.region, rowBuf))
+      case WriteValue(value, pathPrefix, spec) =>
+        val v = emit(value, env)
+        val p = emit(pathPrefix, env)
+
+        val m = mb.newLocal[Boolean]
+        val pv = mb.newLocal[String]
+        val rb = mb.newLocal[OutputBuffer]
+
+        val taskCtx = Code.invokeScalaObject[HailTaskContext](HailTaskContext.getClass, "get")
+        val enc = typeToTypeInfo(value.pType2) match {
+          case BooleanInfo => spec.buildEmitEncoderF[Boolean](value.pType2, mb.fb)
+          case IntInfo => spec.buildEmitEncoderF[Int](value.pType2, mb.fb)
+          case LongInfo => spec.buildEmitEncoderF[Long](value.pType2, mb.fb)
+          case FloatInfo => spec.buildEmitEncoderF[Float](value.pType2, mb.fb)
+          case DoubleInfo => spec.buildEmitEncoderF[Double](value.pType2, mb.fb)
```

Comment:
Do we need `CharInfo`?

Reply:
I accidentally included the change for this while rebasing so it won't show up as a separate commit but---

We don't currently use Char to represent any values, so I hadn't included it, although that doesn't prevent us from introducing that in the future, in addition to things like "Short" and "Byte" etc. I rewrote this in favor of a utility function on the spec that takes the typeInfo and parameterizes on that, which looks much cleaner and avoids having to enumerate all the different types like this.